python/shotgun_fields/image_widget.py

```diff
@@ -258,7 +297,7 @@ def _on_worker_failure(self, uid, msg):
         """
         On failure just display an error and set the toolkit to the error string.
         """
-        if uid == self._task_uid:
+        if str(uid) == self._task_uid:
```

Comment:
I don't remember why I added this... I just tested, and things seem to work without. I'm guessing I was testing/debugging something on a hunch that proved incorrect... I've removed this from both methods.